ProcessMaker/processmaker
# Resolve Loading Screen Issue in My Templates and Public Templates Tabs

## Issue & Reproduction Steps
This PR addresses an issue where the My Template and Public Template tabs consistently display the loading screen on slower networks. This problem arises from the improper implementation of the `apiDataLoading`,` loading`, and `noResults` variables during the component creation, the `fetch` method, and upon receiving a response.

## Solution

- Properly implemented the `apiDataLoading`, `loading`, and `noResults` variables to manage the loading state of the components at various stages.


## How to Test

1. In your browser's network panel, set your connection speed from `No Throttling` to `Slow 3g`.
2. Navigate to the Designer > Screen.
3. Before the page fully loads, select either the My Templates or Public Templates tab.
4. Confirm that the loading screen appears and disappears after a period, indicating successful loading of the templates.

## Related Tickets & Packages
[FOUR-14721](https://processmaker.atlassian.net/browse/FOUR-14721)

ci:next
ci:deploy

## Code Review Checklist


[FOUR-14721]: https://processmaker.atlassian.net/browse/FOUR-14721?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
